components/authz-service/server/v2/project.go

```diff
@@ -158,13 +164,47 @@ func (s *state) UpdateProject(ctx context.Context,
 }
 
 func (s *state) ApplyRulesStart(
-	context.Context, *api.ApplyRulesStartReq) (*api.ApplyRulesStartResp, error) {
-	s.log.Info("apply project rules: START")
-	err := s.projectUpdateManager.Start()
+	ctx context.Context, _ *api.ApplyRulesStartReq) (*api.ApplyRulesStartResp, error) {
+	// NOTE (tc): Only one call to ApplyRulesStart can happen at a time.
+	// This should be good enough to prevent race conditions for single node,
+	// in conjunction with the table locking that happens in store.ApplyStagedRules.
+	// Can still get into a weird state if we panic, but a re-apply will fix things up.
+	// We will be refactoring for multi-node using workflow tooling in the near future.
+	s.applyRuleMux.Lock()
+	defer s.applyRuleMux.Unlock()
```

Comment:
Interesting approach. I suppose we could also have gone with some process (goroutine) reading from a request channel, serving one request at a time, and pushing the response into a response channel. But this is fine, I suppose.